app/controllers/pushes_controller.rb

```diff
@@ -0,0 +1,359 @@
+
+    set_kind_by_tab
+
+    # MIGRATION - ask
+    # Special fix for: https://github.com/pglombardo/PasswordPusher/issues/2811
+    @push.passphrase = ""
+  end
+
+
+  def create
+    # Require authentication if allow_anonymous is false
+    # See config/settings.yml
+    authenticate_user! if Settings.enable_logins && !Settings.allow_anonymous
+
+    @push = Push.new(push_params)
+
+    @push.user_id = current_user.id if user_signed_in?
+
+    create_detect_deletable_by_viewer(@push, push_params)
+    create_detect_retrieval_step(@push, push_params)
+
+    if @push.save
+      log_creation(@push)
+
+      redirect_to preview_push_path(@push)
+    else
+      if @push.kind == "text"
+        @text_tab = true
+      elsif @push.kind == "file"
+        @files_tab = true
+      elsif @push.kind == "url"
+        @url_tab = true
+      else
+        @text_tab = true
+      end
+      render action: "new", status: :unprocessable_entity
+    end
+  end
+
+  def preview
+    @secret_url = helpers.secret_url(@push)
+    @qr_code = helpers.qr_code(@secret_url)
+  end
+
+
+  def print_preview
+    @secret_url = helpers.secret_url(@push)
+    @qr_code = helpers.qr_code(@secret_url)
+
+    @message = print_preview_params[:message]
+    @show_expiration = print_preview_params[:show_expiration]
+    @show_id = print_preview_params[:show_id]
+
+    render action: "print_preview", layout: "naked"
+  end
+
+
+  def preliminary
+    # This password may have expired since the last view.  Validate the password
+    # expiration before doing anything.
+    @push.check_limits
+
+    if @push.expired
+      log_view(@push)
+      render template: "pushes/show_expired", layout: "naked"
+      return
+    else
+      @secret_url = helpers.secret_url(@push, with_retrieval_step: false, locale: params[:locale])
+    end
+
+    render action: "preliminary", layout: "naked"
+  end
+
+  def audit
+    if @push.user_id != current_user.id
+      redirect_to :root, notice: _("That push doesn't belong to you.")
+      return
+    end
+
+    @secret_url = helpers.secret_url(@push)
+  end
+
+
+  def expire
+    # Check if the push is deletable by the viewer or if the user is the owner
+    if @push.deletable_by_viewer == false && @push.user_id != current_user&.id
+      redirect_to :root, notice: _("That push is not deletable by viewers and does not belong to you.")
+      return
+    end
+
+    if @push.expired
+      redirect_to @push, notice: _("That push is already expired.")
+      return
+    end
+
+    @push.expire!
+    log_expire(@push)
+
+    respond_to do |format|
+      format.html { redirect_to @push, notice: t("pushes.expire.expired") }
+    end
+  end
+
+  def index
+    unless Settings.enable_logins
+      redirect_to :root
+      return
+    end
+
+    @filter = params[:filter]
+    allowed_kinds = ["text"]
+    if Settings.enable_file_pushes
+      allowed_kinds << "file"
+    end
+
+    if Settings.enable_url_pushes
+      allowed_kinds << "url"
+    end
```

Comment:
It is possible to limit push kinds by using `Settings.enable_url_pushes` and `Settings.enable_file_pushes`. So, if any of them is disabled temporarily, this kind pushes won't be listed on the dashboard. It is an edge case. So, we can remove this check. @pglombardo what do you think?

Reply:
I like the general idea but people messing with settings will file tickets says "where are my pushes".  Best to show all pushes regardless of the evolution of enabled push types.